qcodes/tests/test_config.py

```diff
@@ -167,102 +164,166 @@ def path_to_config_file_on_disk(tmp_path):
-                     'homedir.')
-    def test_user_schema(self, isfile, load_config, env, schema):
-        schema.return_value = copy.deepcopy(SCHEMA)
-        env.return_value = ENV_KEY
-        isfile.return_value = True
-        load_config.side_effect = partial(side_effect, GOOD_CONFIG_MAP)
-        self.conf.defaults, self.defaults_schema = self.conf.load_default()
-        config = self.conf.update_config()
-        self.assertEqual(config, CONFIG)
-
-    @patch.object(Config, 'current_schema', new_callable=PropertyMock)
-    @patch.object(Config, 'env_file_name', new_callable=PropertyMock)
-    @patch.object(Config, 'load_config')
-    @patch('os.path.isfile')
-    @patch("builtins.open", mock_open(read_data=USER_SCHEMA))
-    def test_bad_user_schema(self, isfile, load_config, env, schema):
-        schema.return_value = copy.deepcopy(SCHEMA)
-        env.return_value = ENV_KEY
-        isfile.return_value = True
-        load_config.side_effect = partial(side_effect, BAD_CONFIG_MAP)
-        with self.assertRaises(jsonschema.exceptions.ValidationError):
-            self.conf.defaults, self.defaults_schema = self.conf.load_default()
-            self.conf.update_config()
-
-    @patch.object(Config, "current_config", new_callable=PropertyMock)
-    def test_update_user_config(self, config):
-        # deep copy because we mutate state
-        config.return_value = copy.deepcopy(CONFIG)
-        self.conf.add("foo", "bar")
-        self.assertEqual(self.conf.current_config, UPDATED_CONFIG)
-
-    @patch.object(Config, 'current_schema', new_callable=PropertyMock)
-    @patch.object(Config, "current_config", new_callable=PropertyMock)
-    def test_update_and_validate_user_config(self, config, schema):
-        self.maxDiff = None
-        schema.return_value = copy.deepcopy(SCHEMA)
-        # deep copy because we mutate state
-        config.return_value = copy.deepcopy(CONFIG)
-        self.conf.add("foo", "bar", "string", "foo", "bar")
-        self.assertEqual(self.conf.current_config, UPDATED_CONFIG)
-        self.assertEqual(self.conf.current_schema, UPDATED_SCHEMA)
+@pytest.fixture(name='config')
+def _make_config():
+    conf = Config()
+    yield conf
+
+
+def test_missing_config_file(config):
+    with pytest.raises(FileNotFoundError):
+        config.load_config("./missing.json")
+
+
+@pytest.mark.skipif(Path.cwd() == Path.home(),
+                    reason="This test requires that "
+                           "working dir is different from homedir.")
+def test_default_config_files(
+        config,
+        mocker
+):
+    schema = mocker.patch.object(
+        Config,
+        'current_schema',
+        new_callable=PropertyMock
+    )
+    env = mocker.patch.object(
+        Config,
+        'env_file_name',
+        new_callable=PropertyMock
+    )
+    load_config = mocker.patch.object(Config, 'load_config')
+    isfile = mocker.patch('os.path.isfile')
```

Comment:
refactor these ~4 things into shared fixture(s)?

Reply:
ok then

Reply:
There is still room for improvements but that will be another time